src/test/hosting/api.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { expect } from "chai";
+import * as nock from "nock";
+import * as sinon from "sinon";
+import * as helpers from "../helpers";
+import * as api from "../../api";
+import * as hostingApi from "../../hosting/api";
+
+const TEST_CHANNELS_RESPONSE = {
+  channels: [
+    // domain exists in TEST_GET_DOMAINS_RESPONSE
+    { url: "https://my-site--ch1-4iyrl1uo.web.app" },
+    // domain does not exist in TEST_GET_DOMAINS_RESPONSE
+    { url: "https://my-site--ch2-ygd8582v.web.app" },
```

Comment:
If this is an existing channel, and it's not in GET_DOMAINS, shouldn't it be added (and be in EXPECTED_DOMAINS)?

Reply:
Not necessarily because this domain could have been added manually via the identity api I just added to show that no new domains get be added during sync, domains are only removed in sync

Reply:
I don't understand, the domain here (`ch2`) is coming from the Channels response, not the identity API. The case I think you described (not removing other domains from the identity API that Hosting doesn't control) would be a domain in GET_DOMAINS _not_ being remove in EXPECTED_DOMAINS, right? (a la `randomurl.com`).

Since `ch2` is a channel that exists (coming in the CHANNELS_RESPONSE) and it is NOT in GET_DOMAINS, I would expect that it would need to be added to the list of authorized domains. (The case here may be that a previous sync failed, or the domain was accidentally removed - the CLI should add it back)

Reply:
So this was an intentional part of the design - if someone manually removes a channel that isn't the current channel being deployed, we should not assume that it was done so accidentally and "correct" it.

If the channel being deployed / created doesn't already have an authorizedDomain, we should add it (because we offer a flag to skip adding it if users desire), but if an unrelated channel is missing its authorizedDomain we should not make assumptions about adding it back. Removing channels domains that no longer exist is safe because we know for a fact the domain doesn't exist and therefore can't break anything the user intended to do.

Reply:
As Micheal said, we do not add any domains during sync we only remove them and that is to respect if the user had manually removed that domain themselves via the console or the identity api

Reply:
The _manually removing a channel from the identity API_ was the part that wasn't coming across. "Sync" to me means that two lists become unified, not that one list is pruned - added a comment to that effect as well. Thanks for all the clarification.